core/src/block.rs

```diff
@@ -434,39 +435,9 @@ mod valid {
             self,
             topology: &Topology,
         ) -> Result<CommittedBlock, (Self, BlockValidationError)> {
-            // TODO: Should the peer that serves genesis have a fixed role of ProxyTail in topology?
-            if !self.payload().header.is_genesis()
-                && topology.is_consensus_required().is_some()
-                && topology
-                    .filter_signatures_by_roles(&[Role::ProxyTail], self.signatures())
-                    .is_empty()
-            {
-                return Err((self, SignatureVerificationError::ProxyTailMissing.into()));
-            }
-
-            #[allow(clippy::collapsible_else_if)]
-            if self.payload().header.is_genesis() {
-                // At genesis round we blindly take on the network topology from the genesis block.
-            } else {
-                let roles = [
-                    Role::ValidatingPeer,
-                    Role::Leader,
-                    Role::ProxyTail,
-                    Role::ObservingPeer,
-                ];
-
-                let votes_count = topology
-                    .filter_signatures_by_roles(&roles, self.signatures())
-                    .len();
-                if votes_count.lt(&topology.min_votes_for_commit()) {
-                    return Err((
-                        self,
-                        SignatureVerificationError::NotEnoughSignatures {
-                            votes_count,
-                            min_votes_for_commit: topology.min_votes_for_commit(),
-                        }
-                        .into(),
-                    ));
+            if !self.payload().header.is_genesis() {
+                if let Err(err) = topology.verify_signatures(self.signatures()) {
```

Comment:
this is ok, although as you can see it does more checks then is strictly required.
If you could I would rather you moved this function into `block.rs` and remove redundant checks